lib/LogCollector.js

```diff
@@ -0,0 +1,207 @@
+/* Copyright @ 2016 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+var Logger = require('./Logger.js');
+
+/**
+ * Stringifies given <tt>object</tt> without any circular references.
+ * @param {object} someObject the <tt>object</tt> to be stringified.
+ * @return {string} the result of <tt>JSON.stringify</tt> with the circular
+ * references skipped.
+ */
+function stringify(someObject){
+    const cache = [];
+    return JSON.stringify(someObject, function (key, value) {
+        if (typeof value === 'object' && value !== null) {
+            if (cache.indexOf(value) !== -1) {
+                // Circular reference
+                return;
+            }
+            cache.push(value);
+        }
+        return value;
+    });
+}
+
+/**
+ * Creates new <tt>LogCollector</tt>. Class implements <tt>LoggerTransport</tt>
+ * and thus can be added as global transport in order to capture all the logs.
+ *
+ * @param {Object} logStorage an object which allows to store the logs collected
+ * @param {function(string)} logStorage.storeLogs a method that will store
+ * the logs which are passed as a JSON string in the following format:
+ * {
+ *   "log1": "logged message1\nlogged message2\n...logged message10\n"
+ * }
+ * Where the number next to the log key indicates log entry number in ascending
+ * order.
+ *
+ * @param {Object} options the <tt>LogCollector</tt> configuration options.
+ * @param {number} options.maxEntryLength the size limit for a single log entry
+ * to be stored. The <tt>LogCollector</tt> will push the entry as soon as it
+ * reaches or exceeds this limit.
+ * @param {number} options.storeInterval how often the logs should be stored in
+ * case <tt>maxEntryLength</tt> was not exceeded.
+ * @param {boolean} options.stringifyObjects indicates whether or not object
+ * arguments should be "stringified" with <tt>JSON.stringify</tt> when a log
+ * message is composed. Note that objects logged on the error log level are
+ * always stringified.
+ *
+ * @constructor
+ */
+function LogCollector(logStorage, options) {
```

Comment:
It feels like it can be useful utility for other people using logger, but I don't mind moving it somewhere else. Should I move it to jitsi-meet ? But then it doesn't fit there very well, because will be used directly from jitsi-meet like the logger is (jitsi-meet-logger is dependency of both jitsi-meet and lib-jitsi-meet).